src/plugins/wizard/public/visualizations/common/expression_helpers.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { cloneDeep } from 'lodash';
+import { OpenSearchaggsExpressionFunctionDefinition } from '../../../../data/public';
+import { ExpressionFunctionOpenSearchDashboards } from '../../../../expressions';
+import { buildExpressionFunction } from '../../../../expressions/public';
+import { VisualizationState } from '../../application/utils/state_management';
+import { getAggService, getIndexPatterns } from '../../plugin_services';
+
+export const getAggExpressionFunctions = async (visualization: VisualizationState) => {
+  const { activeVisualization, indexPattern: indexId = '' } = visualization;
+  const { aggConfigParams } = activeVisualization || {};
+
+  const indexPatternsService = getIndexPatterns();
+  const indexPattern = await indexPatternsService.get(indexId);
+  const aggConfigs = getAggService().createAggConfigs(indexPattern, cloneDeep(aggConfigParams));
+
+  const opensearchDashboards = buildExpressionFunction<ExpressionFunctionOpenSearchDashboards>(
+    'opensearchDashboards',
+    {}
+  );
+
+  // soon this becomes: const opensearchaggs = vis.data.aggs!.toExpressionAst();
+  const opensearchaggs = buildExpressionFunction<OpenSearchaggsExpressionFunctionDefinition>(
+    'opensearchaggs',
+    {
+      index: indexId,
+      metricsAtAllLevels: false,
+      partialRows: false,
```

Comment:
We should probably have a placeholder issue somewhere for supporting partial row aggs.

Reply:
Given that we are migrating away from vislib, I kept the implementation intentionally light so that we have less doc migration issues once we have our final feature set. Im hoping that with this basic version we can get the feedback to identify what those features are, including partial buckets. I'll open a single issue to track all of that :)